### PR TITLE
Fix 22.04 FTBFS

### PR DESCRIPTION
**Pull request by AlanGriffiths**
* Don't use ld, use ldd;
* Don't check .symbols unless using ld; and
* Working incantation for gmock/gtest headers

### CODE REVIEW

File: src/miroil/CMakeLists.txt [hunk 1]
```diff
@@ -61,6 +61,7 @@ set_target_properties(miroil
 
 find_program(MIR_DPKG_GENSYMBOLS dpkg-gensymbols)
 if (CMAKE_COMPILER_IS_GNUCXX   AND  # clang generates slightly different symbols (but we don't care)
+        MIR_USE_LD EQUAL "ld"      AND  # lld on 22.04 generates slightly different symbols (but we don't care)
```
Comment (Saviq) [inline src/miroil/CMakeLists.txt:64]:
```suggestion
    MIR_USE_LD EQUAL "ld"      AND  # lld on 22.04 generates slightly different symbols (but we don't care)
```

File: debian/rules [hunk 2]
```diff
@@ -32,10 +32,9 @@ ifeq ($(filter noopt,$(DEB_BUILD_OPTIONS)),noopt)
 	DEB_BUILD_MAINT_OPTIONS += optimize=-lto
 endif
 
-# Disable LTO on jammy due to failing to build
+# Use ldd on jammy due to failing to build with ld
 ifneq ($(filter jammy,$(DEB_DISTRIBUTION)),)
```
Comment (Saviq) [inline debian/rules:36]:
```suggestion
ifeq ($(DEB_DISTRIBUTION),jammy)
```

**Review by Saviq — CHANGES_REQUESTED** [review]
Couple nits.

File: debian/rules (hunk 2 above)
Comment (AlanGriffiths) [reply debian/rules:36]: Makes sense, but I wanted to only touch the minimum.

File: src/miroil/CMakeLists.txt (hunk 1 above)
Comment (AlanGriffiths) [reply src/miroil/CMakeLists.txt]: Fixed

**Review by Saviq — APPROVED** [review]
Yes.

bors merge

**Comment by AlanGriffiths** [comment]
OK, so `debian/compat` can't contain 9 anymore. Just need to see what we can bump it to, and what changes there are...

**Comment by AlanGriffiths** [comment]
OK, toolchain is beyond workarounds

**Pull request closed without merging**